src/components/SingleArticle.jsx

```diff
@@ -1,29 +1,49 @@
 import { useParams } from 'react-router-dom';
 import { useEffect, useState } from 'react';
-import { getArticle } from '../utils/api';
+import { getArticle, patchVotes } from '../utils/api';
+import { Button } from 'react-bootstrap';
 
 const SingleArticle = () => {
     const { article_id } = useParams();
     const [singleArticle, setSingleArticle] = useState({});
+    const [votes, setVotes] = useState(0);
 
     useEffect(() => {
         getArticle(article_id)
             .then((article) => {
                 setSingleArticle(article);
+                setVotes(article.votes);
             })
             .catch((err) => {
                 console.log(err);
             })
     }, [article_id]);
 
+    const updateVotes = (vote) => {
+        patchVotes(article_id, {inc_votes: vote})
```

Comment:
The votes should be optimistally rendered here. Take your setVotes out of the promise chain with the API call and then the votes will change even while the api call is still being made. If you want to be thorough, then you could put a catch block on the patchVotes promise which will decrement the vote in the even it rejects.